packages/quantic/force-app/main/default/lwc/quanticSearchInterface/quanticSearchInterface.js

```diff
@@ -30,21 +40,30 @@ export default class QuanticSearchInterface extends LightningElement {
 
   connectedCallback() {
     loadDependencies(this).then((CoveoHeadless) => {
-      getHeadlessConfiguration().then((data) => {
-        if (data) {
-          this.engineOptions = {
-            configuration: {
-              ...JSON.parse(data),
-              search: {
-                searchHub: this.searchHub,
-                pipeline: this.pipeline
-              }
-            }
-          };
-          setEngineOptions(this.engineOptions, CoveoHeadless.buildSearchEngine, this.engineId, this);
-          setInitializedCallback(this.initialize, this.engineId);
-        }
-      });
+      if (!getHeadlessBindings(this.engineId).engine) {
+        getHeadlessConfiguration().then((data) => {
+          if (data) {
+            this.engineOptions = {
+              configuration: {
+                ...JSON.parse(data),
+                search: {
+                  searchHub: this.searchHub,
+                  pipeline: this.pipeline,
+                },
+              },
+            };
+            setEngineOptions(
+              this.engineOptions,
+              CoveoHeadless.buildSearchEngine,
+              this.engineId,
+              this
+            );
```

Comment:
Maybe we can put this in a private that returns a promise ?

Reply:
Pas fou ca. Would clean it up a bit

Reply:
Not sure I understand what you are suggesting, which could be returning a promise?

Reply:
I think the suggestion is just extracting this getting and setting of options into a private function. Could improve legibility.

Reply:
more improving readability than legibility but yeah

Reply:
Yes wrong word. My bad